libs/wingsdk/src/shared-gcp/counter.inflight.ts

```diff
@@ -0,0 +1,60 @@
+import { Datastore } from "@google-cloud/datastore";
+import type { ICounterClient } from "../cloud";
+
+// const AMOUNT_TOKEN = "amount";
+// const INITIAL_VALUE_TOKEN = "initial";
+const COUNTER_ID = "counter";
+// const VALUE_ATTRIBUTE = "counter_value";
+// const SET_VALUE = "set_value";
+
+export class CounterClient implements ICounterClient {
+  private readonly client: Datastore;
+
+  constructor(
+    private readonly databaseName: string,
+    private readonly initial: number = 0
+  ) {
+    this.client = new Datastore({ databaseId: this.databaseName });
+  }
+
+  public async inc(amount = 1, key = COUNTER_ID): Promise<number> {
+    const kind = "Counter";
+    const name = key;
+    const counterKey = this.client.key([kind, name]);
+
+    // Fetch the current counter
+    const [existingCounter] = await this.client.get(counterKey);
+
+    // Determine the current value of the counter or use the initial value
+    const currentValue = existingCounter?.data?.count || this.initial;
+
+    // Calculate the new value
+    const newValue = currentValue + amount;
+
+    // Save the updated counter
+    const counter = {
+      key: counterKey,
+      data: {
+        count: newValue,
+      },
+    };
+
+    await this.client.save(counter);
+    console.log(`Saved ${counter.key.name}: ${counter.data.count}`);
```

Comment:
I'm not sure we need this log

Reply:
Just a debugging log since it's all still WIP.